app/src/main/java/org/kiwix/kiwixmobile/webserver/ZimHostActivity.java

```diff
@@ -172,17 +174,18 @@ public void open(BooksOnDiskListItem.BookOnDisk bookOnDisk) {
   }
 
   public void select(BooksOnDiskListItem.BookOnDisk bookOnDisk) {
-    List<BooksOnDiskListItem> selectedBooks = null;
-    for (int i = 0; i < booksAdapter.getItemCount(); i++) {
-      if (booksAdapter.getItems().get(i).equals(bookOnDisk)) {
-        booksAdapter.getItems().get(i).setSelected(true);
-        selectedBooks.add(booksAdapter.getItems().get(i));
+    ArrayList<BooksOnDiskListItem> booksList = new ArrayList<>();
```

Comment:
it is a bit neater to do 
```
    ArrayList<BooksOnDiskListItem> booksList = new ArrayList<>();
    for (BooksOnDiskListItem item : booksAdapter.getItems()) {
      if (item.equals(bookOnDisk)) {
        item.setSelected(!item.isSelected())
      }
      booksList.add(item)
    }
    booksAdapter.setItems(booksList);
  }
```
if you are incrementing an index variable you should not be using a `for-each` and instead using a normal for. `for-each` looks a lot better though

Reply:
`booksList.add(item)`
This will add every item user has interacted with i.e. selected/unselected at the end of the list. After that you're setting `booksAdapter` with this list that has at first all the items of `booksAdapter` and then the items user has interacted with.
But what we want is a list that would just replace the old states i.e. `isSelected` with new ones. I'm afraid your code doesn't implements the functionality we want.

Reply:
it does exactly the same.
`booksList.addAll(booksAdapter.getItems());`
that adds everything to booklist
`booksList.get(i).setSelected(!bookOnDisk.isSelected());`
this selects them.

Instead I have contained all logic to the loop
`item.setSelected(!item.isSelected())` select
`booksList.add(item)` add

Reply:
Here, `booksList.add(item)` you're only adding the books that user has interacted with.
And when you `booksAdapter.setItems(booksList)` do this after the loop, `booksAdapter` will only contain books the user has interacted with?
What about the books that user didn't interacted with?

Reply:
Ah, gotcha.

Reply:
I am not. I add every single item that is in the adapter.
```
    ArrayList<BooksOnDiskListItem> booksList = new ArrayList<>(); //create empty array
    for (BooksOnDiskListItem item : booksAdapter.getItems()) {
      if (item.equals(bookOnDisk)) { //if this item is the selected item
        item.setSelected(!item.isSelected()) //select it
      }
      booksList.add(item) //add the item, wether it is selected or not
    }
    booksAdapter.setItems(booksList); //set the adapter with the new list
  }

Reply:
Implementing this now. :thumbsup:

Reply:
Fixed in ca12e1791026832b17dfd4bc412faeed84f11a8b

Reply:
Fixed. ca12e17